packages/eslint-config-next/package.json

```diff
@@ -12,6 +12,7 @@
   "dependencies": {
     "@next/eslint-plugin-next": "15.0.0-canary.7",
     "@rushstack/eslint-patch": "^1.3.3",
+    "@typescript-eslint/eslint-plugin": "^6.0.0 || 7.0.0 - 7.2.0",
```

Comment:
Why not `^7.0.0`?

Reply:
Heh, because when I'd first sent this PR, 6 was the latest version. And I wasn't sure what the support story for `eslint-config-next` is. Since the `canary` branch still uses 6.x:

https://github.com/JoshuaKGoldberg/next.js/blob/e296846427aefb59ba1ff9445b1c9ee4743960e2/package.json#L115

...I figured I'd leave in both versions. 🤷

Reply:
That part makes sense. It's more about `7.0.0 - 7.2.0` vs `^7.0.0`. Why would a minor eslint-plugin release break something?

Reply:
Oh, sorry, got it. This is how the `canary` branch pins `@typescript-eslint/parser`:

https://github.com/vercel/next.js/blob/12dbc8ad7ee107f7a55bb7e7f75ecfd50ecf6b14/packages/eslint-config-next/package.json#L15

...which I don't see any reason for. I'd +1 going with `^7.0.0` in theory.

Reply:
Is there a bug in typescript-eslint, then, that should be fixed? I don't believe we intentionally meant to raise the minimum Node version in 7.3 - that would be something we'd save for a major version. We'd consider raising a dependency's minimum version to one that requires a higher Node.js version to be that kind of major breaking change.

We generally do want folks to use the same versions of all `@typescript-eslint/*` packages - which is part of why for flat config there's a single `typescript-eslint` package that can be used.

Can I lean on you to touch this up as you'd prefer? 😅

Reply:
I think this is just an oversight on our part. The engine restriction was changed in 7.3. but it was just a formality: https://github.com/typescript-eslint/typescript-eslint/pull/8671.

So if you're on Node.js 18.17, linting was already defect. You might just not know about it.

I'll relax the version in a follow-up since we can upgrade our own engines in 15.x.